RecoJets/JetProducers/plugins/FixedGridRhoProducerFastjetFromRecHit.cc

```diff
@@ -35,9 +37,10 @@ class FixedGridRhoProducerFastjetFromRecHit : public edm::stream::EDProducer<> {
   static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
 
 private:
+  void beginRun(edm::Run const &, edm::EventSetup const &) override;
   void produce(edm::Event &, const edm::EventSetup &) override;
   std::array<double, 4> getHitP4(const DetId &detId, const double hitE, const CaloGeometry &caloGeometry) const;
-  bool passedHcalNoiseCut(const HBHERecHit &hit) const;
+  bool passedHcalNoiseCut(const HBHERecHit &hit, float thres) const;
```

Comment:
what about passing the entire `HcalPFCuts` pointer here and rely on being non-null to decide the logic?